### ISSUE TITLE
QDrawer ignores 'show-if-above' property when window is resized

### ISSUE THREAD

**Issue by patryk-tech** [quasarframework/quasar#6905]
**Describe the bug**
Opening a window less than breakpoint size wide (`< 1024px`) and resizing it above the breakpoint size makes drawers appear, even when they don't have the `show-if-above` property.

When opening a window that is greater than 1024 px wide, only the drawers with the property are shown, which means the behaviour is inconsistent.

**Codepen/jsFiddle/Codesandbox (required)**
I can't get the drawers to work on CodePen, but I have made a quick project that reproduces the error [in this Gitlab repository](https://gitlab.com/patryk.tech/bugs/-/tree/quasar-drawers-show-if-above) - with the bug occurring in [MainLayout.vue](https://gitlab.com/patryk.tech/bugs/-/blob/quasar-drawers-show-if-above/quasar/src/layouts/MainLayout.vue).

**To Reproduce**
1. Open a browser window less than 1024px wide.
2. Go to https://bugs.patryk.tech/#/
3. Resize the window so it is more than 1024 pixels wide.
4. As you can see, both drawers are now visible.

**Expected behavior**
Only drawers with the `show-if-above` property should be open, just as when opening a new window.

**Screenshots**
![qbug01](https://user-images.githubusercontent.com/47309236/80599667-f533cf80-8a33-11ea-9475-2dc11b328f9e.png)
![qbug02](https://user-images.githubusercontent.com/47309236/80599669-f664fc80-8a33-11ea-865e-e4fa76f5f044.png)
![qbug03](https://user-images.githubusercontent.com/47309236/80599673-f6fd9300-8a33-11ea-9d99-c98ced71d8c8.png)

**Platform (please complete the following information):**
OS: Arch Linux
Node: v12.16.2
Browsers: Firefox 75.0 (Linux, 64-bit), Google Chrome 81.x (Windows 10)

**Comment by pdanpdan** [comment]
Hello.

```
Force drawer to be shown on screen on initial render if the layout width is above breakpoint, regardless of v-model; This is the default behavior when SSR is taken over by client on initial render
```

**Comment by patryk-tech** [comment]
> Force drawer to be shown on screen **on initial render** if the layout width is above breakpoint, regardless of v-model; This is the default behavior when SSR is taken over by client on initial render

Could you elaborate?

My problem isn't with the initial render, as that works fine. The problem only occurs after resizing.

Even if I set the default values to `null` in `data`, then explicitly to `false` in `mounted`, resizing the window still makes them visible, which IMO it should not do.

**Comment by rstoenescu** [comment]
@patryk-tech 

Please use the appropriate channel to request help. [Discord](https://chat.quasar.dev) is more appropriate.

This is in no way a bug as you haven't grasped the docs yet. Quasar is equipped with any behavior that you want for QDrawer. Have you looked at the QDrawer's API card in docs? Notice the `breakpoint` prop. Configure it accordingly.

**Comment by patryk-tech** [comment]
> This is in no way a bug as you haven't grasped the docs yet.

@rstoenescu with all due respect, I disagree, but I may of course be missing something.

> Quasar is equipped with any behavior that you want for QDrawer. Have you looked at the QDrawer's API card in docs? Notice the `breakpoint` prop. Configure it accordingly.

I have. Maybe something wasn't clear to me, but from my understanding, the `breakpoint` should be setting the drawer's classes, not the `model` attribute.

If I set `:breakpoint="0"`, it doesn't change the model's value, but it renders it as a desktop drawer, even on mobile. Conversely, if I set it to an arbitrarily large value (e.g. `:breakpoint="1000000"`), it will always render in mobile mode.

The inconsistency is lies in the fact that resizing the window toggles the `model`, not the `class` attributes. IMHO, it should respect `show-if-above` when you resize it, otherwise, you get two different outcomes for very similar situations.

You open a 1024 pixel window, and

**Comment by patryk-tech** [comment]
> This is in no way a bug as you haven't grasped the docs yet.

@rstoenescu with all due respect, I disagree, but I may of course be missing something.

> Quasar is equipped with any behavior that you want for QDrawer. Have you looked at the QDrawer's API card in docs? Notice the `breakpoint` prop. Configure it accordingly.

I have. Maybe something wasn't clear to me, but from my understanding, the `breakpoint` should be setting the drawer's classes, not the `model` attribute.

If I set `:breakpoint="0"`, it doesn't change the model's value, but it renders it as a desktop drawer, even on mobile. Conversely, if I set it to an arbitrarily large value (e.g. `:breakpoint="1000000"`), it will always render in mobile mode.

The inconsistency is lies in the fact that resizing the window toggles the **model**, not the *class* attributes. IMHO, it should respect `show-if-above` when you resize it, otherwise, you get two different outcomes for very similar situations.

- You open a 1024 pixel window, and the right-hand drawer is not rendered, because the initial render is `> breakpoint`. Resizing the window below 1024px wide doesn't affect the `:model` at all.
- You open a 1023 pixel window, and make it 1px wider. It toggles the `:model` property, even though it does not have a `show-if-above` property.

> Please use the appropriate channel to request help. Discord is more appropriate.

I will swing by discord in a bit, but I sincerely believe this behaviour is **inconsistent**, and that *should* classify it as a bug.

**Comment by rstoenescu** [comment]
The `breakpoint` prop does not mean it sets a bunch of classes on it and that's it. It influences the behavior of QDrawer (the limit between mobile and desktop mode): 1. when you resize the window width (switches to/from mobile mode) and 2. `show-if-above` prop makes use of the breakpoint to determine the initial render state -- aka "do I render it in mobile or desktop mode".

As the docs clearly state, `show-if-above` influences only the initial render and is there as a helper. Its purpose ends after the initial render.

If you need your drawers in a certain state, simply use the v-model with the correct initial assignment (true/false). You can even initialize the model based on the Screen plugin (leftDrawer: this.$q.screen.width > 1000).

**Comment by patryk-tech** [comment]
> If you need your drawers in a certain state, simply use the v-model with the correct initial assignment (true/false). You can even initialize the model based on the Screen plugin (leftDrawer: this.$q.screen.width > 1000).

My problem isn't with the initial state. It's that *after* I resize the window, quasar sets the model to `true` **if** the initial window width was `<1024`. I don't think it should do that, otherwise you get inconsistencies.

If I set the models to `false` in `data()`...

no.|Initial width|show-if-above|Initial model|Model after resizing < 1024|Model after resizing > 1024
--|--|--|--|--|--
1|1023|**false**|false|false|**true**
2|1023|true|false|false|true
3|1024|false|false|false|false
4|1024|true|true|false|true

That's not me setting the model to `true`, that is Quasar, and I don't think it should (on line no.1). It's inconsistent with behaviour on line no. 3.

**Comment by pdanpdan** [comment]
Line 1: width change from below to above breakpoint => false to true
Line 3: width change from above to above breakpoint => no change

**Comment by patryk-tech** [comment]
@pdanpdan I realize that. But if initial size is 1024, then you resize it below breakpoint (500), then resize it above the breakpoint, it doesn't change the model.

Resize from 1920 to 500 => no change.
Resize from 500 to 1920 => no change (from below to above breakpoint).

I still think it should respect the `show-if-above` property, otherwise your state is difficult to predict. What if your user opens the site on a tablet, and changes from portrait to landscape? Or starts their browser in an arbitrary size, and they maximize it?

**Comment by pdanpdan** [comment]
Please specify the exact conditions to reproduce.

**Comment by rstoenescu** [comment]
Again, the `show-if-above` prop works only for the initial render like explained earlier. Its purpose ends after the initial render.

Anyway, QDrawer remembers the last state when it was above the breakpoint. If you start the page with above the breakpoint but the model is false (and show-if-above is false) then any resize will keep it this way. But if you start with any combination of model and show-if-above value, then resizing to small then to big means "I don't have a previous 'big' state to remember, now I'm over the breakpoint, so then I show the drawer". This is because you don't want the user to close the drawer then to resize window (big to small and to big again) and the drawer to pop out.

What you are suggesting is a way for you to tell the QDrawer not to toggle state by itself. Correct?

**Comment by patryk-tech** [comment]
> "I don't have a previous 'big' state to remember, now I'm over the breakpoint, so then I show the drawer".

> What you are suggesting is a way for you to tell the QDrawer not to toggle state by itself. Correct?

Really, the only time I have found this to be an issue is when starting below the breakpoint width, and resize it to >1024 px.

My use case is that I have one left drawer for navigation (with `show-if-above`) and a right drawer for settings (without). I don't feel the settings drawer should open unless the user explicitly clicks it, so *ideally* I'd love for Quasar to say "I don't have a previous big state to remember, now I'm over the breakpoint, so I check whether this drawer has `show-if-above` and show it." Then it would get the same state regardless whether the window was resized, or already above the breakpoint on initial load.

If you think adding another property would be easier/more logical, I trust your judgment... You are far more familiar with the codebase than I am :)

**Comment by rstoenescu** [comment]
New behavior will be available in v1.10.0.

**Comment by patryk-tech** [comment]
Thank you both!
